composer.json

```diff
@@ -47,6 +48,11 @@
     "branch-alias": {
       "dev-main": "1.9.4.x-dev"
     },
-    "magento-root-dir": "root"
+    "magento-root-dir": "."
+  },
+  "config": {
+    "allow-plugins": {
+      "magento-hackathon/magento-composer-installer": true
+    }
```

Comment:
I can't find a way to install the module (and update the composer.lock) without these lines. I also think that magento-root-dir should really be "."

What do you guys think? Is it ok? Could you give me a hand here?

Reply:
I recommend to test it very good what a root-dir of "." will do with an install of this project. Iam not sure what kind of unintended consequences we can have.

Reply:
I don't understand (and never found any doc) how you can have a composer json/lock without those lines. composer requires them. How was the composer json/lock generated if composer doesn't work without those?

Reply:
I've manually removed the line, although everything seems very weird about this composer.json

Reply:
> I don't understand (and never found any doc) how you can have a composer json/lock without those lines. composer requires them. How was the composer json/lock generated if composer doesn't work without those?

simply said, you dont. The composer.json was intended for having OpenMage as a dependency, not as an installable project. So all this questions were not relevant till now.